src/Idris/REPL.idr

```diff
@@ -644,6 +644,63 @@ loadMainFile f
            [] => pure (FileLoaded f)
            _ => pure (ErrorsBuildingFile f errs)
 
+
+nfun : {auto c : Ref Ctxt Defs} ->
+  {vs : _} ->
+  REPLEval -> Defs -> Env Term vs -> Term vs -> Core (Term vs)
+nfun NormaliseAll = normaliseAll
+nfun _ = normalise
+
+||| Infer the type of a PTerm and elaborate it to a Term
+||| Returns a pair containing the term and its type
+inferAndElab : {auto c : Ref Ctxt Defs} ->
+  {auto u : Ref UST UState} ->
+  {auto s : Ref Syn SyntaxInfo} ->
+  {auto m : Ref MD Metadata} ->
+  {auto o : Ref ROpts REPLOpts} ->
+  ElabMode ->
+  PTerm ->
+  Core (Term [], Term [])
```

Comment:
Can we get a `TermAndType` record type so that these can be accessed by name?
They have exactly the same type so it's very easy to use the wrong one.